### PR TITLE
Update go module against k8s.io/cloud-provider

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Update go module against k8s.io/cloud-provider

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
Update go module against k8s.io/cloud-provider
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,34 +3,75 @@ module sigs.k8s.io/cloud-provider-azure
...
+	k8s.io/legacy-cloud-providers v0.18.5
+	k8s.io/utils v0.0.0-20200619165400-6e3d28b6ed19
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.10 // indirect
 )
 
 replace (
 	bitbucket.org/bertimus9/systemstat => bitbucket.org/bertimus9/systemstat v0.0.0-20180207000608-0eeff89b0690
 	cloud.google.com/go => cloud.google.com/go v0.38.0
-	github.com/Azure/azure-sdk-for-go => github.com/Azure/azure-sdk-for-go v40.2.0+incompatible
+	github.com/Azure/azure-sdk-for-go => github.com/Azure/azure-sdk-for-go v44.0.1-0.20200710185145-8277be38f5a2+incompatible
```
Comment (andyzhangx) [inline go.mod:74]: in Line6 it's `v44.0.0`, inconsistent with this line.
Pls also paste all command you used for this go mod change.

Comment (nilo19) [reply go.mod:74]: just `go get -u ./...` after manually modify some lines, for example, link k8s.ui/cloud-provider from k/staging to external repo.

Comment (nilo19) [reply go.mod:74]: > just `go get -u ./...` after manually modify some lines, for example, link `k8s.io/cloud-provider` from `k/staging` to external repo.

Comment (andyzhangx) [reply go.mod:74]: what about: 
edit go.mod
```console
go mod tidy
go mod vendor
```

Comment (nilo19) [reply go.mod:74]: yeah these two commands are executed after updating `go/mod`.

Comment (andyzhangx) [reply go.mod:74]: usually I won't run `go get -u ./...` after go.mod change, only
```
#edit go.mod
go mod tidy
go mod vendor
```

Comment (nilo19) [reply go.mod:32]: got it, updated.

**Pull request merged**